cla_public/apps/checker/fields.py

```diff
@@ -161,10 +161,22 @@ def __init__(self, label=None, validators=None, min_val=0, max_val=9999999999, *
         self.min_val = min_val
         self.max_val = max_val
 
+    def convert_amount_to_utf8(self, amount):
+        # Covert pounds side of the input into UTF-8 format
+        try:
+            amount = amount.decode("utf-8")
+        except UnicodeEncodeError:
+            # Input is already in UTF-8 format
+            pass
+
+        return amount
+
     def process_formdata(self, valuelist):
         if valuelist:
             pounds, _, pence = valuelist[0].strip().partition(".")
-            pounds = re.sub(r"[\s,]+", "", pounds)
+            pounds = self.convert_amount_to_utf8(pounds)
```

Comment:
From the commit message, I get the impression this was moved out to a method in order to satisfy a cyclomatic complexity check more than anything else. While I agree some of this code could be moved out, I think the simple utf-decode-unless-exception just doesn't warrant moving out, as it's plenty simple enough to encounter inline. If it truly is some reusable bit of logic, I think it'd be better suited in some "text utils" area of the codebase than still within the field class.